src/dataflow/components/nodes/factories/data-storage-rete-node-factory.tsx

```diff
@@ -77,8 +79,45 @@ export class DataStorageReteNodeFactory extends DataflowReteNodeFactory {
   private addInputs(node: Node) {
     const lastKey = Array.from(node.inputs.keys())[node.inputs.size - 1];
     const lastInput = node.inputs.get(lastKey);
-    if (lastInput && lastInput.connections.length > 0) {
+    if (lastInput && lastInput.connections.length > 0 && this.editor) {
       // final input has a connection, add an additional input
+      const name = lastInput.connections[0].output.node?.name;
+      let sequenceName = "my-sequence";
+      if (name === "Number" || name === "Timer") {
+        sequenceName = name;
+      } else if (name === "Sensor") {
+        if (lastInput.connections[0].output.node) {
+          const nodeType = String(lastInput.connections[0].output.node.data.type);
+          const sensorType = NodeSensorTypes.find((s: any) => s.type === nodeType);
+          sequenceName = nodeType === "none" ? name : (sensorType ? sensorType.name : sequenceName);
+        }
+      } else if (name === "Generator") {
+        if (lastInput.connections[0].output.node) {
+          sequenceName = String(lastInput.connections[0].output.node.data.generatorType);
+        }
+      } else if (name === "Math") {
+        if (lastInput.connections[0].output.node) {
+          sequenceName = String(lastInput.connections[0].output.node.data.mathOperator);
+        }
+      } else if (name === "Logic") {
+        if (lastInput.connections[0].output.node) {
+          sequenceName = String(lastInput.connections[0].output.node.data.logicOperator);
+        }
+      } else if (name === "Transform") {
+        if (lastInput.connections[0].output.node) {
+          sequenceName = String(lastInput.connections[0].output.node.data.transformOperator);
+        }
```

Comment:
Might be better to refactor this into a function in `node.ts` - that's where this sort of stuff lives.